WC_Layers/Source_Mesh_Reader.py

```diff
@@ -0,0 +1,49 @@
+import h5py
+import numpy as np
+import yaml
+import argparse
+    
+def extract_source_data(source_mesh_list, num_elements, photon_groups):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    
+    input: iterable of .h5m filenames (str), whose files contain photon source information
+    output: numpy array of source density data with rows = # of mesh elements and columns = # number of photon groups, with one array per source mesh
+    ''' 
+    sd_list = np.ndarray((len(source_mesh_list), num_elements, photon_groups))
+    for source_index, source_name in enumerate(source_mesh_list):
+         file = h5py.File(source_name, 'r')
+         sd_list[source_index,:] = file['tstt']['elements']['Tet4']['tags']['source_density'][:]

```

Comment:
Probably good to have a comment before this:
```suggestion
         # directly accessing the data stored in the HDF5 as specified by the MOAB mesh library
         # MOAB data is stored in the `tstt` group, and this data is in a `tag` with name `source_density`
         # applied to each of the `Tet4` type of `elements`
         sd_list[source_index,:] = file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```